src/Microsoft.Graph.Core/Requests/BatchRequestBuilder.cs

```diff
@@ -99,5 +103,26 @@ public async Task<RequestInformation> ToPostRequestInformationAsync(BatchRequest
             requestInfo.Headers.Add("Content-Type", CoreConstants.MimeTypeNames.Application.Json);
             return requestInfo;
         }
+
+        private static async Task ThrowIfFailedResponseAsync(HttpResponseMessage httpResponseMessage)
+        {
+            if (httpResponseMessage.IsSuccessStatusCode) return;
+
+            if (httpResponseMessage is { Content.Headers.ContentType.MediaType: string contentTypeMediaType } && contentTypeMediaType.StartsWith(CoreConstants.MimeTypeNames.Application.Json, StringComparison.OrdinalIgnoreCase))
+            {
+                using var responseContent = await httpResponseMessage.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                using var document = await JsonDocument.ParseAsync(responseContent).ConfigureAwait(false);
```

Comment:
sorry for the late feedback on this. We should conditionally pass a cancellation token for both methods here (and below as well)